### PR TITLE
adding linkerd and namerd packages

### PR DESCRIPTION
**Pull request by siggy**
This introduces linkerd and namerd packages to the universe. Both are open source packages, built from https://github.com/BuoyantIO/linkerd

Thanks for all the ongoing support, excited to get this out there!

A couple notes:

1) ~~We are preparing a new release this week, and would like to rev the docker version numbers before this merges to `version-2.x`.~~

2) ~~We were unable to get the service URL working in the linkerd package. We are hoping to link to our admin page via something like `http://mesos-url/service/linkerd`. Please have a look at our use of ports, along with the following labels:~~
- `DCOS_PACKAGE_FRAMEWORK_NAME`
- `DCOS_SERVICE_NAME`
- `DCOS_SERVICE_SCHEME`
- `DCOS_SERVICE_PORT_INDEX`

### CODE REVIEW

File: repo/packages/L/linkerd/0/config.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+          "pattern": "^(?:\\.[a-z][a-z0-9]*?(?:-[a-z0-9]+)*)+$"
+        },
+        "framework-name": {
+          "default": "linkerd",
+          "description": "The framework name to register with Mesos.",
+          "type": "string",
+          "pattern": "^/?(([a-z0-9]|[a-z0-9][a-z0-9\\-]*[a-z0-9])\\.)*([a-z0-9]|[a-z0-9][a-z0-9\\-]*[a-z0-9])$"
+        },
+        "instances": {
+          "default": 1,
```
Comment (pyronicide) [inline repo/packages/L/linkerd/0/config.json]: Set this to 1000, it'll deploy to everything then.

File: repo/packages/L/linkerd/0/marathon.json.mustache [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  "maintainer": "hello@buoyant.io",
+  "container": {
+    "type": "DOCKER",
+    "docker": {
+      "image": "{{resource.assets.container.docker.linkerd-docker}}",
+      "network": "BRIDGE",
+      "privileged": true,
+      "portMappings": [
+        {
+          "containerPort": 0,
```
Comment (pyronicide) [inline repo/packages/L/linkerd/0/marathon.json.mustache]: Normally, you want the containerPort to be fixed. Are you sure?

File: repo/packages/L/linkerd/0/marathon.json.mustache [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+          "hostPort": {{linkerd.admin-port}},
+          "servicePort": 0,
+          "protocol": "tcp"
+        },
+        {
+          "containerPort": 0,
```
Comment (pyronicide) [inline repo/packages/L/linkerd/0/marathon.json.mustache]: Normally, you want the containerPort to be fixed. Are you sure?

File: repo/packages/L/linkerd/0/marathon.json.mustache [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  },
+  "healthChecks": [
+    {
+      "protocol": "HTTP",
+      "portIndex": 0,
+      "path": "/admin/ping"
+    }
+  ],
+  "args": [
+    "/mnt/mesos/sandbox/{{linkerd.config-filename}}"
```
Comment (pyronicide) [inline repo/packages/L/linkerd/0/marathon.json.mustache]: I was wondering how you were using that. Cool!

File: repo/packages/L/linkerd/0/marathon.json.mustache [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  ],
+  "env": {
+    "FRAMEWORK_NAME":"{{linkerd.framework-name}}",
+    "WEBURI": "http://{{linkerd.framework-name}}{{linkerd.dns-suffix}}:{{linkerd.admin-port}}"
+  },
+  "uris": [
+    "{{linkerd.config-uri}}/{{linkerd.config-filename}}"
+  ],
+  "labels": {
+    "DCOS_PACKAGE_FRAMEWORK_NAME": "{{linkerd.framework-name}}",
```
Comment (pyronicide) [inline repo/packages/L/linkerd/0/marathon.json.mustache]: This doesn't get used anywhere. The `DCOS_SERVICE_*` ones are the important ones.

File: repo/packages/L/linkerd/0/package.json [hunk 6]
```diff
@@ -0,0 +1,20 @@
+{
+  "packagingVersion": "2.0",
+  "name": "linkerd",
+  "version": "0.3.0-SNAPSHOT-0.1",
+  "framework": true,
```
Comment (pyronicide) [inline repo/packages/L/linkerd/0/package.json]: Pretty sure you're not using a mesos scheduler, right?

File: repo/packages/L/linkerd/0/package.json [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  "scm": "https://github.com/BuoyantIO/linkerd/",
+  "description": "A dynamic linker for microservices.",
+  "maintainer": "hello@buoyant.io",
+  "website": "https://linkerd.io",
+  "tags": ["linkerd", "namerd", "buoyant", "loadbalancer", "service-discovery", "proxy", "microservices", "rpc"],
+  "preInstallNotes": "We recommend a minimum of 0.25 CPUs and 256 MB of RAM available for linkerd.",
```
Comment (pyronicide) [inline repo/packages/L/linkerd/0/package.json]: Probably want something about needing to run on every agent.

File: repo/packages/N/namerd/0/marathon.json.mustache [hunk 8]
```diff
@@ -0,0 +1,63 @@
+{
+  "id": "{{namerd.name}}",
+  "instances": {{namerd.instances}},
+  "cpus": {{namerd.cpus}},
+  "mem": {{namerd.mem}},
+  "acceptedResourceRoles": [ "{{namerd.resource-role}}" ],
```
Comment (pyronicide) [inline repo/packages/N/namerd/0/marathon.json.mustache]: Is there a reason someone would want `namerd` on a `slave_public`?

File: repo/packages/N/namerd/0/marathon.json.mustache [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+  "maintainer": "hello@buoyant.io",
+  "container": {
+    "type": "DOCKER",
+    "docker": {
+      "image": "{{resource.assets.container.docker.namerd-docker}}",
+      "network": "BRIDGE",
+      "privileged": true,
+      "portMappings": [
+        {
+          "containerPort": 0,
```
Comment (pyronicide) [inline repo/packages/N/namerd/0/marathon.json.mustache]: Normally, you want the containerPort to be fixed. Are you sure?

File: repo/packages/N/namerd/0/marathon.json.mustache [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+    {{namerd.admin-port}},
+    {{namerd.http-port}},
+    {{namerd.thrift-port}}
+  ],
+  "env": {
+    "FRAMEWORK_NAME":"{{namerd.framework-name}}",
+    "WEBURI": "http://{{namerd.framework-name}}{{namerd.dns-suffix}}:{{namerd.admin-port}}"
+  },
+  "uris": [
+    "{{namerd.config-uri}}/{{namerd.config-filename}}"
```
Comment (pyronicide) [inline repo/packages/N/namerd/0/marathon.json.mustache]: This is going to work a little oddly if it isn't in resources. I'm not sure of a better way to do it, just will require extra configuration for those who's DCs are disconnected.

File: repo/packages/N/namerd/0/marathon.json.mustache (hunk 8 above)
Comment (siggy) [inline repo/packages/N/namerd/0/marathon.json.mustache]: Possible. namerd provides a REST API to control naming configuration across your cluster. It persists this data in ZK. We've found it convenient to have a namerd on `slave_public` to demonstrate API usage.

File: repo/packages/L/linkerd/0/config.json (hunk 1 above)
Comment (siggy) [inline repo/packages/L/linkerd/0/config.json]: we thought about doing this. this leaves linkerd stuck at `Waiting` in the marathon ui. if there's no workaround for that, we may switch this back to `1`, as it makes it look like linkerd is not correctly installed.

File: repo/packages/L/linkerd/0/marathon.json.mustache (hunk 2 above)
Comment (siggy) [inline repo/packages/L/linkerd/0/marathon.json.mustache]: i'm not sure. everything seems to work correctly with `"containerPort": 0,`. i'm testing with fixed now.

File: repo/packages/L/linkerd/0/config.json (hunk 1 above)
Comment (pyronicide) [inline repo/packages/L/linkerd/0/config.json]: There's no workaround unfortunately. Message that they have to change it in the post-install notes?

File: repo/packages/N/namerd/0/marathon.json.mustache (hunk 8 above)
Comment (pyronicide) [inline repo/packages/N/namerd/0/marathon.json.mustache]: :+1:

**Comment by siggy** [comment]
@pyronicide i just bumped the docker images versions to our latest release. we are good to go on our end. let me know if there are any more changes you'd like to see. thanks!

**Comment by mhausenblas** [comment]
@siggy can you please merge latest changes from `version-2.x` into your branch so that we can merge this?

**Comment by siggy** [comment]
@mhausenblas i just merged the latest changes from `version-2.x`. let me know if you need anything else. thanks!

**Pull request merged**